### PR TITLE
executor: migrate test-infra to testify for stale_txn_test

### LINKED ISSUE #28618: migrate test-infra to testify for `stale_txn_test.go`

### PR DESCRIPTION
**Pull request by hawkingrei**
Signed-off-by: Weizhen Wang <wangweizhen@pingcap.com>

### What problem does this PR solve?

Issue Number: close #28618

Problem Summary:

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

**Review by tisonkun — APPROVED** [review]

**Comment by hawkingrei** [comment]
@disksing @tangenta PTAL

File: executor/stale_txn_test.go [hunk 1]
```diff
@@ -68,30 +69,33 @@ func (s *testStaleTxnSerialSuite) TestExactStalenessTransaction(c *C) {
...
+	store, clean := testkit.CreateMockStore(t)
+	defer clean()
+	tk := testkit.NewTestKit(t, store)
 	tk.MustExec("use test")
 	for _, testcase := range testcases {
-		c.Log(testcase.name)
-		failpoint.Enable("tikvclient/injectTxnScope",
-			fmt.Sprintf(`return("%v")`, testcase.zone))
+		t.Log(testcase.name)
+		failpoint.Enable("tikvclient/injectTxnScope", fmt.Sprintf(`return("%v")`, testcase.zone))
```
Comment (wshwsh12) [inline executor/stale_txn_test.go:78]: require.NoError(failpoint.Enablexxxx)

File: executor/stale_txn_test.go [hunk 2]
```diff
@@ -68,30 +69,33 @@ func (s *testStaleTxnSerialSuite) TestExactStalenessTransaction(c *C) {
...
-			startTS := oracle.ExtractPhysical(tk.Se.GetSessionVars().TxnCtx.StartTS)
-			c.Assert(curTS-startTS, Less, time.Second.Milliseconds())
-			c.Assert(startTS-curTS, Less, time.Second.Milliseconds())
+			startTS := oracle.ExtractPhysical(tk.Session().GetSessionVars().TxnCtx.StartTS)
+			require.Less(t, curTS-startTS, time.Second.Milliseconds())
+			require.Less(t, startTS-curTS, time.Second.Milliseconds())
 		}
 		tk.MustExec("commit")
 	}
 	failpoint.Disable("tikvclient/injectTxnScope")
```
Comment (wshwsh12) [inline executor/stale_txn_test.go:92]: ditto.

File: executor/stale_txn_test.go [hunk 3]
```diff
@@ -540,42 +554,42 @@ func (s *testStaleTxnSerialSuite) TestSetTransactionReadOnlyAsOf(c *C) {
...
+		require.Equal(t, testcase.expectedTS, tk.Session().GetSessionVars().TxnCtx.StartTS)
 		tk.MustExec("commit")
-		c.Assert(tk.Se.GetSessionVars().TxnReadTS.PeakTxnReadTS(), Equals, uint64(0))
+		require.Equal(t, uint64(0), tk.Session().GetSessionVars().TxnReadTS.PeakTxnReadTS())
 		tk.MustExec("begin")
-		c.Assert(tk.Se.GetSessionVars().TxnCtx.StartTS, Not(Equals), testcase.expectedTS)
+		require.NotEqual(t, tk.Session().GetSessionVars().TxnCtx.StartTS, testcase.expectedTS)
 		tk.MustExec("commit")
 
 		failpoint.Disable("github.com/pingcap/tidb/expression/injectSafeTS")
```
Comment (wshwsh12) [inline executor/stale_txn_test.go:570]: ditto

File: executor/stale_txn_test.go [hunk 4]
```diff
@@ -1016,16 +1043,18 @@ func (s *testStaleTxnSerialSuite) TestStaleReadPrepare(c *C) {
 
 	// test prepared stale select in stale txn
 	tk.MustExec(fmt.Sprintf(`start transaction read only as of timestamp '%s'`, time1.Format("2006-1-2 15:04:05.000")))
-	c.Assert("execute p1", NotNil)
+	require.NotNil(t, "execute p1")
```
Comment (wshwsh12) [inline executor/stale_txn_test.go:1046]: tk.MustExec("execute p1")???

File: executor/stale_txn_test.go [hunk 5]
```diff
@@ -1016,16 +1043,18 @@ func (s *testStaleTxnSerialSuite) TestStaleReadPrepare(c *C) {
...
 	tk.MustExec("commit")
 
 	// assert execute prepared statement should be error after set transaction read only as of
 	tk.MustExec(fmt.Sprintf(`set transaction read only as of timestamp '%s'`, time1.Format("2006-1-2 15:04:05.000")))
-	c.Assert("execute p1", NotNil)
+	require.NotNil(t, "execute p1")
```
Comment (wshwsh12) [inline executor/stale_txn_test.go:1051]: ditto

File: executor/stale_txn_test.go (hunk 3 above)
Comment (hawkingrei) [reply executor/stale_txn_test.go:570]: Done

File: executor/stale_txn_test.go (hunk 4 above)
Comment (hawkingrei) [reply executor/stale_txn_test.go]: Done

File: executor/stale_txn_test.go (hunk 5 above)
Comment (hawkingrei) [reply executor/stale_txn_test.go]: Done

File: executor/stale_txn_test.go (hunk 2 above)
Comment (hawkingrei) [reply executor/stale_txn_test.go]: Done

File: executor/stale_txn_test.go (hunk 1 above)
Comment (hawkingrei) [reply executor/stale_txn_test.go]: done

**Review by wshwsh12 — APPROVED** [review]
LGTM

**Pull request merged**